frontend/src/components/LoginAndRegistration.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from 'react';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import Nav from 'react-bootstrap/Nav';
+import Tab from 'react-bootstrap/Tab';
+import Tabs from 'react-bootstrap/Tabs';
+import Container from 'react-bootstrap/Container';
+import TabContainer from 'react-bootstrap/TabContainer';
+import Login from './Login';
+import Registration from './Registration';
+import style from './LoginAndRegistration.module.css';
+import { useLocation } from 'react-router-dom'
+
+function useQuery() {
+  return new URLSearchParams(useLocation().search);
+}
+
+export default () => {
+  const [ tokenObj, setTokenObj ] = useState({});
+  const [ userObj, setUserObj ] = useState({});
+
+  let query = useQuery();
+  let code = query.get("code");
+
+  function fetchTokens() {
```

Comment:
This component contains functions that are either very similar or the same as the ones above. Maybe putting it in a separate API file is an even better idea. :)

Reply:
@naijauser You created the `fetchTokens` function, but you didn't actually use it here.